### PR TITLE
feat: ordering of Flex's EdgeValue fields

### PR DESCRIPTION
**Pull request by LG95**
### Purpose
Ordered the fields of Flex's EdgeValues according to their priorities.

### CODE REVIEW

File: backend/sample/spring/src/main/kotlin/br/com/zup/beagle/sample/spring/service/TestService.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * limitations under the License.
+ */
+
+package br.com.zup.beagle.sample.spring.service
+
+import br.com.zup.beagle.sample.builder.TestScreenBuilder
+import org.springframework.stereotype.Service
+
+@Service
+class TestService {
```
Comment (LG95) [inline backend/sample/spring/src/main/kotlin/br/com/zup/beagle/sample/spring/service/TestService.kt:23]: Please use a clearer name, like `SampleFlexService`.

File: backend/sample/core/src/main/kotlin/br/com/zup/beagle/sample/builder/TestScreenBuilder.kt [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+                    text = "3",
+                    alignment = TextAlignment.CENTER
+                ).applyFlex(flex = Flex(
+                    size = Size(width = 33.33.unitPercent(),height = 50.unitReal()))
+                ).applyStyle(style = Style(backgroundColor = "#649d66"))
+            )
+        ).applyFlex(flex = Flex(
+            flexDirection = FlexDirection.ROW
+        )).applyStyle(style = Style(backgroundColor = "#000000"))
+    )
```
Comment (LG95) [inline backend/sample/core/src/main/kotlin/br/com/zup/beagle/sample/builder/TestScreenBuilder.kt:83]:
```suggestion
        child = Container(
            children = listOf(
                this.createText("1").applyFlex(flex = Flex(
                    positionType = FlexPositionType.ABSOLUTE,
                    position = Position(all = 50.unitReal(), horizontal = 100.unitReal(), top = 45.unitReal()),
                    size = Size(width = 33.33.unitPercent(), height = 50.unitReal()))
                ).applyStyle(style = Style(backgroundColor = "#142850")),
                this.createText("2").applyFlex(flex = Flex(
                    size = Size(width = 33.33.unitPercent(), height = 50.unitReal()))
                ).applyStyle(style = Style(backgroundColor = "#dd7631")),
                this.createText("3").applyFlex(flex = Flex(
                    size = Size(width = 33.33.unitPercent(), height = 50.unitReal()))
                ).applyStyle(style = Style(backgroundColor = "#649d66"))
            )
        ).applyFlex(flex = Flex(
            flexDirection = FlexDirection.ROW
        )).applyStyle(style = Style(backgroundColor = "#000000"))
    )

    private fun createText(s: String) = Text(text = s, alignment = TextAlignment.CENTER)
```

**Comment by tuliopereirazup** [comment]
What happens if I pass this attribute in different order when writing the json manually?

**Comment by LG95** [comment]
@tuliopereirazup well then web will differ from mobile in their priorites for these fields, since their priority is based on order and they are counting on the BFF for ordering.

**Comment by tuliopereirazup** [comment]
> @tuliopereirazup well then web will differ from mobile in their priorites for these fields, since their priority is based on order and they are counting on the BFF for ordering.

Then I think this should be handled on front-end or should be well documented.

**Comment by LG95** [comment]
@pedropnaves what do you think?

**Comment by pedropnaves** [comment]
> @pedropnaves what do you think?

I think we should document both this and the order of priority in the mobiles.

**Comment by uziasferreirazup** [comment]
I really disagree with this approach of having this type of rule in json, mainly because it doesn't offer a bff for all languages, what the problem handle this in the front-end?

**Comment by LG95** [comment]
This will be done in the web framework.

**Pull request closed without merging**